quartz-core/src/main/java/org/quartz/simpl/RAMJobStore.java

```diff
@@ -665,6 +665,32 @@ public TriggerState getTriggerState(TriggerKey triggerKey) throws JobPersistence
 
     /**
      * <p>
+     * Reset the current state of the identified <code>{@link Trigger}</code>.
+     * </p>
+     *
+     * @see TriggerState#NORMAL
+     * @see TriggerState#PAUSED
+     * @see TriggerState#COMPLETE
+     * @see TriggerState#ERROR
+     * @see TriggerState#BLOCKED
+     * @see TriggerState#NONE
+     */
+    public void resetTriggerState(TriggerKey triggerKey) throws JobPersistenceException {
+        synchronized (lock) {
+            TriggerWrapper tw = triggersByKey.get(triggerKey);
+
+            // does the trigger exist?
+            if (tw == null || tw.trigger == null) {
+                return;
+            }
+            tw.state = TriggerWrapper.STATE_WAITING;
+            applyMisfire(tw);
```

Comment:
on point 1.  I agree the call is redundant/superfluous.  no real harm, no real gain.

on point 2. I agree that consistency would be good - it may make a difference at some point in the future.